test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file initFileTransfer_Http_harness.c
+ * @brief Implements the proof harness for initFileTransfer_Http function.
+ */
+/* Http interface includes. */
+#include "ota_http_private.h"
+
+/* Stub required for the proof. */
+OtaHttpStatus_t init( char * pUrl )
+{
+    OtaHttpStatus_t status;
+
+    return status;
+}
+
+void initFileTransfer_Http_harness()
+{
+    OtaAgentContext_t * pAgentCtx;
+    OtaFileContext_t fileContext;
+    OtaHttpInterface_t http;
+    OtaHttpStatus_t status;
+    OtaInterfaces_t interface;
+    OtaAgentContext_t agent;
+
+    pAgentCtx = &agent;
+
+    pAgentCtx->fileContext = fileContext;
+
+    /* The init reference in the http interface will always point to a function address and thus
```

Comment:
How do we know this to be the case?

Reply:
I have added an assert statement to check if the function pointer is initialized or not.